compiler/hash-messaging/src/lib.rs

```diff
@@ -1 +1,48 @@
 //! Defines compiler messages that are passed in and out of the compiler.
+
+pub mod listener;
+pub mod stream;
+
+use hash_pipeline::settings::CompilerSettings;
+use hash_reporting::report::Report;
```

Comment:
We need to figure out how to not expose compiler internals here...

Reply:
Yep, we definitely need to decide what to keep internal or external!